<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from /home/daniell/dev/robotics/ros2_iiwa_controller/src/iiwa_description/urdf/iiwa7.urdf.xacro | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="iiwa7">
  <link name="world"/>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_0">
    <inertial>
      <origin rpy="0 0 0" xyz="-0.012857 0.0 0.069964"/>
      <mass value="4.855658"/>
      <inertia ixx="0.017839" ixy="0.0" ixz="0.000781" iyy="0.022294" iyz="0.0" izz="0.021334"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_0.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_0.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A0" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="lbr_iiwa7_link_0"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_1">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.034819 0.123299"/>
      <mass value="3.394011"/>
      <inertia ixx="0.021383" ixy="0.0" ixz="0.0" iyy="0.020403" iyz="0.003797" izz="0.007563"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.1475"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_1.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.1475"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_1.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A1" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.1475"/>
    <parent link="lbr_iiwa7_link_0"/>
    <child link="lbr_iiwa7_link_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="700" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="1.710422666954443"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_2">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.039793 0.086944"/>
      <mass value="4.031991"/>
      <inertia ixx="0.031697" ixy="0.0" ixz="0.0" iyy="0.03008" iyz="0.005889" izz="0.009666"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0105 -0.34"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_2.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0105 -0.34"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_2.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A2" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 -0.0105 0.1925"/>
    <parent link="lbr_iiwa7_link_1"/>
    <child link="lbr_iiwa7_link_2"/>
    <axis xyz="0 1 0"/>
    <limit effort="700" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="1.710422666954443"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_3">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.034819 0.123299"/>
      <mass value="3.394011"/>
      <inertia ixx="0.021383" ixy="0.0" ixz="0.0" iyy="0.020403" iyz="-0.003797" izz="0.007563"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.5475"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_3.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.5475"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_3.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A3" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0105 0.2075"/>
    <parent link="lbr_iiwa7_link_2"/>
    <child link="lbr_iiwa7_link_3"/>
    <axis xyz="0 0 1"/>
    <limit effort="700" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="1.7453292519943295"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_4">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.039793 0.086944"/>
      <mass value="4.031989"/>
      <inertia ixx="0.031695" ixy="0.0" ixz="0.0" iyy="0.030079" iyz="-0.005889" izz="0.009665"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 -0.0105 -0.74"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_4.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 -0.0105 -0.74"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_4.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A4" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0105 0.1925"/>
    <parent link="lbr_iiwa7_link_3"/>
    <child link="lbr_iiwa7_link_4"/>
    <axis xyz="0 -1 0"/>
    <limit effort="700" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="2.2689280275926285"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_5">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 -0.029824 0.076267"/>
      <mass value="1.529239"/>
      <inertia ixx="0.008485" ixy="0.0" ixz="0.0" iyy="0.007136" iyz="0.002806" izz="0.003848"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.9475"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_5.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 -0.9475"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_5.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A5" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 -0.0105 0.2075"/>
    <parent link="lbr_iiwa7_link_4"/>
    <child link="lbr_iiwa7_link_5"/>
    <axis xyz="0 0 1"/>
    <limit effort="700" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="2.443460952792061"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_6">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0.07102 0.00495"/>
      <mass value="2.403626"/>
      <inertia ixx="0.007067" ixy="0.0" ixz="0.0" iyy="0.006804" iyz="0.000311" izz="0.004629"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0707 -1.14"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_6.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0707 -1.14"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_6.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A6" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 -0.0707 0.1925"/>
    <parent link="lbr_iiwa7_link_5"/>
    <child link="lbr_iiwa7_link_6"/>
    <axis xyz="0 1 0"/>
    <limit effort="700" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="3.141592653589793"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <!-- ========== LINK ========== -->
  <link name="lbr_iiwa7_link_7">
    <inertial>
      <origin rpy="0 0 0" xyz="3.0000e-06 -2.0000e-06 1.3782e-02"/>
      <mass value="0.259474"/>
      <inertia ixx="0.000171" ixy="0.0" ixz="0.0" iyy="0.000171" iyz="0.0" izz="0.000299"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0.0 0.0 -1.231"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/visual/link_7.dae" scale="1 1 1"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 -1.231"/>
      <geometry>
        <mesh filename="package://iiwa_description/meshes/collision/link_7.stl" scale="1 1 1"/>
      </geometry>
    </collision>
  </link>
  <!-- ========== JOINT ========== -->
  <joint name="lbr_iiwa7_A7" type="revolute">
    <origin rpy="0 0 0" xyz="0.0 0.0707 0.091"/>
    <parent link="lbr_iiwa7_link_6"/>
    <child link="lbr_iiwa7_link_7"/>
    <axis xyz="0 0 1"/>
    <limit effort="700" lower="-3.0543261909900763" upper="3.0543261909900763" velocity="3.141592653589793"/>
    <dynamics damping="10.0" friction="0.1"/>
  </joint>
  <joint name="lbr_iiwa7_joint_ee" type="fixed">
    <parent link="lbr_iiwa7_link_7"/>
    <child link="lbr_iiwa7_link_ee"/>
    <origin rpy="0 0 0" xyz="0 0 0.035"/>
  </joint>
  <link name="lbr_iiwa7_link_ee"/>
  <ros2_control name="lbr_iiwa7_controller" type="system">
    <!-- <hardware>
                <plugin>mock_components/GenericSystem</plugin>
                <param name="calculate_dynamics">true</param>
                <param name="robot_ip">172.31.1.148</param>
                <param name="robot_port">30005</param>
            </hardware> -->
    <hardware>
      <plugin>iiwa_controller/IIWAHardwareInterface</plugin>
      <param name="simulate">False</param>
      <param name="robot_ip">192.168.21.147</param>
      <param name="robot_port">30200</param>
    </hardware>
    <joint name="lbr_iiwa7_A1">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-2.9670597283903604</param>
      <param name="max">2.9670597283903604</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="lbr_iiwa7_A2">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-2.0943951023931953</param>
      <param name="max">2.0943951023931953</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="lbr_iiwa7_A3">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-2.9670597283903604</param>
      <param name="max">2.9670597283903604</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="lbr_iiwa7_A4">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-2.0943951023931953</param>
      <param name="max">2.0943951023931953</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="lbr_iiwa7_A5">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-2.9670597283903604</param>
      <param name="max">2.9670597283903604</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="lbr_iiwa7_A6">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-2.0943951023931953</param>
      <param name="max">2.0943951023931953</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
    <joint name="lbr_iiwa7_A7">
      <param name="initial_position">0.0</param>
      <command_interface name="position"/>
      <param name="min">-3.0543261909900763</param>
      <param name="max">3.0543261909900763</param>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
      <state_interface name="effort"/>
    </joint>
  </ros2_control>
</robot>
